src/lrtest.jl

```diff
@@ -111,37 +112,48 @@ function lrtest(mods::StatisticalModel...; atol::Real=0.0)
     Δdf = _diff(df)
     dfr = Int.(dof_residual.(mods))
 
-    if (forward && any(x -> x > 0, Δdev)) || (!forward && any(x -> x < 0, Δdev))
-        throw(ArgumentError("Residual deviance must be strictly lower " *
-                            "in models with more degrees of freedom"))
+    ll = loglikelihood.(mods)
+    chisq = (NaN, 2 .* abs.(_diff(ll))...)
+
+    for i in 2:length(dev)
+        if ((forward && dev[i-1] < dev[i]) ||
+            (!forward && dev[i-1] > dev[i])) &&
+            dev[i-1] ≉ dev[i]
```

Comment:
Should the user be able to pass a tolerance value for the approximate equality here?

Also, I don't yet fully understand why (and it could be that I've done something wrong), but this still seems not to be true for beta regression. As an example, using the data defined [here](https://github.com/ararslan/BetaRegression.jl/blob/main/test/runtests.jl#L42-L80), I get
```julia
julia> model = fit(BetaRegressionModel, @formula((food / income) ~ 1 + income + persons), data);

julia> nullmodel = fit(BetaRegressionModel, @formula((food / income) ~ 1), data);

julia> (deviance(model), dof(model))
(37.18054389739939, 4)

julia> (deviance(nullmodel), dof(nullmodel))
(36.36829121985548, 2)
```
Compare that to R:
```r
> library(betareg)
> data(FoodExpenditure, package="betareg")
> model <- betareg(I(food / income) ~ 1 + income + persons, data=FoodExpenditure, link="logit", link.phi="identity")
> nullmodel <- betareg(I(food / income) ~ 1, data=FoodExpenditure, link="logit", link.phi="identity")
> sum(residuals(model, type="deviance")^2)
37.18054
> nobs(model) - df.residual(model)
4
> sum(residuals(nullmodel, type="deviance")^2)
36.36829
> nobs(nullmodel) - df.residual(nullmodel)
2
```

Reply:
Quoting [myself](https://github.com/JuliaStats/StatsModels.jl/issues/260#issuecomment-1202888917):

> Part of the fundamental difficulty with mixed models -- both computationally and in terms of users' understanding -- is the way foundational regression theory is taught. Many equalities that hold for a particular subset of the exponential family are taught as the actual definitions of things like deviance, residual degrees of freedom, etc., instead of special cases. (@ararslan you've also felt this with beta regression!)
> 
> Addressing @ararslan's comment: I don't remember the default `isnested` definition, but maybe the "residual deviance must be strictly lower in models with more degrees of freedom" check should be part of `isnested` instead of `lrtest` itself. Or maybe some other function that can have methods defined for new model types without defining needing to specialize `lrtest`.

Reply:
`isnested` has no fallback definition. I think I added that check on the deviance being smaller as a sanity check for models that do not define `isnested` (yet). I could disable it when `isnested` is defined, trusting implementations to enforce it. But results would become meaningless if that check doesn't hold (at least with some tolerance), so keeping it would make sense. It could be based on the log-likelihood rather than the deviance though, so that its relation to the test statistic is more direct.

Regarding beta regression, I'm not sure why the null model has a lower deviance. Maybe that's due to the fact that they have a different dispersion, so that deviance should be scaled accordingly? At least it has a lower log-likelihood, which is reassuring.

Maybe we should stop reporting the deviance altogether when printing `LRTestResult` objects, and instead show the likelihood? R's `anova(..., test="LRT")` prints the deviance, but it uses the scaled deviance to perform the test, which can be misleading (the scaling factor isn't printed...). We could keep the internal field for backward compatibility and only change the printing.

Reply:
Since the test statistic is using loglikelihood, it makes more sense (to me) to report it in the output (instead of deviance). MATLAB has a quite verbose output when comparing models. 

```

    Theoretical Likelihood Ratio Test

    Model    DF    AIC       BIC       LogLik     LRStat     deltaDF    pValue 
    lm1      4     44.602    55.023    -18.301                                 
    lm2      5     45.952    58.978    -17.976    0.65029    1          0.42001
```

Reply:
I've also checked Stata's `lrtest` and R's `lmtest::lrtest`, they also report similar statistics as those reported by MATLAB. So we can probably drop the deviance. I'm not sure about the AIC and the BIC (there's also the AICc...).

Reply:
Let's print both the log-likelihood and the deviance for now to limit disruption for users. I've updated the check that more complex models have a lower deviance to use the log-likelihood instead.

Reply:
Updating to use log likelihood instead of deviance means BetaRegression now Just Works™ with `lrtest` and produces results identical to R's `lrtest`. 🥳 Thanks Milan!